packages/api/src/controllers/asset.ts

```diff
@@ -169,49 +173,43 @@ app.get("/", authMiddleware({}), async (req, res) => {
 });
 
 app.get("/:id", authMiddleware({}), async (req, res) => {
-  const os = await db.asset.get(req.params.id);
-  if (!os) {
-    res.status(404);
-    return res.json({
-      errors: ["not found"],
-    });
+  const asset = await db.asset.get(req.params.id);
+  if (!asset) {
+    throw new NotFoundError(`Asset not found`);
   }
 
-  if (req.user.admin !== true && req.user.id !== os.userId) {
-    res.status(403);
-    return res.json({
-      errors: ["user can only request information on their own assets"],
-    });
+  if (req.user.admin !== true && req.user.id !== asset.userId) {
+    throw new ForbiddenError(
+      `      errors: ["user can only request information on their own assets"],
+      `
+    );
   }
 
-  res.json(os);
+  res.json(asset);
 });
 
-// TODO: Delete this API? Assets will only be created by task result events.
-app.post(
-  "/",
-  authMiddleware({ anyAdmin: true }),
-  validatePost("asset"),
-  async (req, res) => {
-    const id = uuid();
-    const playbackId = await generateUniquePlaybackId(req.store, id);
-    const doc = await validateAssetPayload(
-      id,
-      playbackId,
-      req.user.id,
-      Date.now(),
-      req.config.vodObjectStoreId,
-      req.body
-    );
-    if (!req.user.admin) {
-      res.status(403);
-      return res.json({ errors: ["Forbidden"] });
-    }
-    await db.asset.create(doc);
-    res.status(201);
-    res.json(doc);
+app.post("/export", authMiddleware({}), async (req, res) => {
+  const assetId = req.body.assetId;
+  const asset = await db.asset.get(assetId);
+  if (!asset) {
+    throw new NotFoundError(`Asset not found with id ${assetId}`);
+  }
+  if (req.user.id !== asset.userId) {
+    throw new ForbiddenError(`User can only export their own assets`);
   }
-);
+
+  let task = await req.taskScheduler.scheduleTask(asset, "export", {
```

Comment:
Nice `scheduleTask` helper 😌 

I think the `update` to `phase: "waiting"` should be inside it as well though.